packages/core/src/components/Button/Button.tsx

```diff
@@ -116,21 +138,24 @@ export const HvButton = forwardRef<HTMLButtonElement, HvButtonProps>(
       }
     };
 
+    const StyledComponent = useMemo(() => StyledButton(Component), [Component]);
+
     return (
-      <StyledButton
+      <StyledComponent
         id={id}
-        className={clsx(className, classes?.root, buttonClasses.root)}
         ref={ref}
+        className={clsx(className, classes?.root, buttonClasses.root)}
         onClick={onClick}
         disabled={disabled}
+        aria-disabled={disabled}
```

Comment:
Is this necessary? 🤔 
We already have `disabled`, and [they're semantically the same](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled#description). The difference is that `disabled` does "more stuff", which we actually do want

Reply:
Okay, I will remove this

Reply:
I added this property because not all HTML elements have this `disabled` property. For example, if my root node is a `a` element, I need the `aria-disabled` property to tell the accessibility tree that the element is disabled.